### PR TITLE
Feature/docker api implementation

### PR DESCRIPTION
**Pull request by 24santoshr**
Please review the request and let me know for any changes. 
Also, please run the below command to set up the local docker engine host in command prompt. The code uses localhost 0.0.0.0:9095

edit /lib/systemd/system/docker.service 
ExecStart=/usr/bin/dockerd -H fd:// -H=tcp://0.0.0.0:9095 
systemctl daemon-reload 
sudo service docker restart

### CODE REVIEW

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala [hunk 1]
```diff
@@ -114,23 +142,325 @@ class RequestHandler (configuration: Configuration) extends AppLogging {
...
+    if (!instanceDao.hasInstance(id)) {
+      OperationResult.IdUnknown
+    } else if (!isInstanceDockerContainer(id)) {
+      OperationResult.NoDockerContainer
+    } else {
+      val instance = instanceDao.getInstance(id).get
+      if (instance.instanceState == InstanceState.Running) {
+        log.info(s"Handling /pause for instance with id $id...")
+        //  dockerActor ! PauseMessage(instance.dockerId.get, instanceHasState())
+        //TODO: execute pause command (async?)
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala]: Is is not possible to pause containers with the api we are using?

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala [hunk 2]
```diff
@@ -114,23 +142,325 @@ class RequestHandler (configuration: Configuration) extends AppLogging {
...
+  def handleResume(id: Long): OperationResult.Value = {
+    if (!instanceDao.hasInstance(id)) {
+      OperationResult.IdUnknown
+    } else if (!isInstanceDockerContainer(id)) {
+      OperationResult.NoDockerContainer
+    } else {
+      val instance = instanceDao.getInstance(id).get
+      if (instance.instanceState == InstanceState.Paused) {
+        log.info(s"Handling /resume for instance with id $id...")
+        //TODO: Pause the container (async?)
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala]: Same as above, is it not possible to resume containers ?

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/container.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package de.upb.cs.swt.delphi.instanceregistry.Docker
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/container.scala]: Please rename the file to start with a capital letter (_Container.scala_).

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/exceptions.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package de.upb.cs.swt.delphi.instanceregistry.Docker
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/exceptions.scala]: Please rename the file to start with a capital letter.

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/image.scala [hunk 5]
```diff
@@ -0,0 +1,81 @@
+package de.upb.cs.swt.delphi.instanceregistry.Docker
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/image.scala]: Please rename the file to start with a capital letter.

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/DockerActor.scala [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  implicit val system: ActorSystem = Registry.system
+  implicit val materializer: ActorMaterializer = Registry.materializer
+  implicit val ec: ExecutionContext = system.dispatcher
+  val container = new ContainerCommands(connection)
+
+  case class DockerClient(connection: DockerConnection)
+
+
+  log.info(s"DockerActor started")
```
Comment (johannesduesing) [inline src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/DockerActor.scala:23]: Is it possible to do an initial connection test to see whether the specified docker host is reachable ? Something like getting the version of the docker service would be ideal. If the docker host is not reachable, we should terminate the application.

**Review by johannesduesing — CHANGES_REQUESTED** [review]

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala (hunk 1 above)
Comment (24santoshr) [reply src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala]: I have implemented the pause functionality now.

File: src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/DockerActor.scala (hunk 6 above)
Comment (24santoshr) [reply src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/DockerActor.scala:23]: Yes, it should be possible. Will work on it.

Comment (johannesduesing) [reply src/main/scala/de/upb/cs/swt/delphi/instanceregistry/Docker/DockerActor.scala:23]: As discussed, this is not needed anymore, as all docker commands implement proper error handling and we do not want the application to terminate when docker is not reachable.

**Review by johannesduesing — DISMISSED** [review]

**Review by johannesduesing — APPROVED** [review]

**Pull request merged**